### PR TITLE
Event detail page

### PR DESCRIPTION
**Pull request by AndrewHocking**
Related to #7 

- Implemented Event detail page
- Added more accessibility filters to event occurrences
- Updated event creation form to match
- Fixed faulty route URLs so they accept navigating there with and without a / at the end
- Moved template macros into one `macros.html` file because it was a little confusing having them all over the place (idk if this is the right move, feel free to chime in)
- Made tags clickable on event detail page, linking to a search page with that tag pre-filled

### CODE REVIEW

**Comment by AndrewHocking** [comment]
Known issue: I'm getting an angry warning from SQLAlchemy. I think there is some sort of naming conflict. Everything seems to work as expected currently though, so I'll figure out what's causing it and fix it later.

File: website/templates/event-details.html
Comment (prabjot09) [inline website/templates/event-details.html:1]: I think line 140 has a typo with the time formal where the `l` characters should be replaced with `I`.

**Comment by prabjot09** [comment]
> Known issue: I'm getting an angry warning from SQLAlchemy. I think there is some sort of naming conflict. Everything seems to work as expected currently though, so I'll figure out what's causing it and fix it later.

SQLAlchemy seems requires a different structure for many-to-many relationships with associative fields: https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#combining-association-object-with-many-to-many-access-patterns. Basically, we need additional fields for the EventContributer (the association object) in the User and Event tables, then the `user` and `event` fields in EventContributer would populate those instead. I implemented something similar to EventContributers for the notifications as well, so I can merge it into your changes in a pull request afterwards and you can review it.

**Review by ryan-pelchat — APPROVED** [review]

**Comment by AndrewHocking** [comment]
Oh awesome, thanks Prabjot

**Pull request merged**